### PR TITLE
[RF] Fix gcc9 warnings.

### PR DESCRIPTION
**Pull request by hageboeck**
- Fixing gcc9 warnings by either defaulting, implementing or deleting assignment operators.
- Implementing the assignment operator of RooAbsData required removing members that were
iterators. They have been replaced by range-based for loops.
- This entailed cleaning of RooDataHist code.

ROOT-10135

### CODE REVIEW

**Comment by hageboeck** [comment]
@phsft-bot build just on fedora30/default

**Comment by hageboeck** [comment]
@phsft-bot build just on fedora30

**Comment by hageboeck** [comment]
@phsft-bot build just on fedora30/default

**Comment by amadio** [comment]
@phsft-bot build just on ROOT-fedora30/cxx17

**Comment by hageboeck** [comment]
Building on fedora30 resulted in 50 warnings, but none were in roofit. Previously, a full CI build also passed all tests.

File: roofit/histfactory/inc/RooStats/HistFactory/Data.h [hunk 1]
```diff
@@ -27,10 +27,12 @@ class Data {
   //friend class Channel;
 
   Data();
-  Data( const Data& other );
+  Data( const Data& other ) = default;
```
Comment (Axel-Naumann) [inline roofit/histfactory/inc/RooStats/HistFactory/Data.h]: Can we not simply remove the declarations of copy ctor and assignment op?

Comment (hageboeck) [reply roofit/histfactory/inc/RooStats/HistFactory/Data.h]: Sure, sometimes you don't see the simple things.

Comment (Axel-Naumann) [reply roofit/histfactory/inc/RooStats/HistFactory/Data.h]: I?! ;-)))

**Comment by Axel-Naumann** [comment]
@phsft-bot build

**Comment by Axel-Naumann** [comment]
Also merged into v6-18-00-patches as b6fec4f440a6bb7b3b6cadc226da0e4bed0ae19b and 58c755f078ef36588a14629072d45d8ae8822ece

**Pull request merged**